flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowSerializationSchema.java

```diff
@@ -360,7 +360,8 @@ private SerializationRuntimeConverter assembleRowConverter(
 		return (mapper, reuse, object) -> {
 			ObjectNode node;
 
-			if (reuse == null) {
+			// reuse could be a NullNode if last record is null.
+			if (!(reuse instanceof ObjectNode)) {
```

Comment:
Have you considered to use `if (reuse == null || reuse.isNull())` which might be more straightforward?

Reply:
Yes, I write it like `if (reuse == null || !(reuse instanceof ObjectNode))` for the first version. But the IDE suggests to remove `reuse == null` because `instanceof` can cover that.
However, your way seems more straight forward, and the IDE won't complain about it. I'll change it this way.